### PR TITLE
Do not delete templated CRDs

### PR DESCRIPTION
**Pull request by PhilGrayson**
Fixes issue #7279.

Prevent the deletion of CRDs that were defined in the `templates/`
directory. This makes CRD deletion behaviour consistent with Helm
documentation:

 > CRDs are never deleted. Deleting a CRD automatically deletes all of the
 > CRD’s contents across all namespaces in the cluster. Consequently, Helm
 > will not delete CRDs.

Previous the documentation only applied to CRDs that were defined in the
`crds/` directory. It did not consider that Charts could have CRDs in the
`templates/` directory (for example charts that were written before the
`crds/` directory feature or if the Chart author needed templated CRDs).

Signed-off-by: Phil Grayson <phil@philgrayson.com>

### CODE REVIEW

**Comment by PhilGrayson** [comment]
If there is support for this feature I'll have a go at implementing tests.

**Comment by thomastaylor312** [comment]
Good catch on this @PhilGrayson! So my one concern is around people who may use CRDs in the template directory without them being the old Helm 2 style "crd-install" hooks. Deleting those if you delete the chart could be a desired behavior. I am pinging some of the other maintainers to get their opinions on it as well. 

One possible solution would be to only ignore those that have the `crd-install` hook annotation, but I wouldn't necessarily start down that path until we figure out what the best behavior would be

**Review by thomastaylor312 — APPROVED** [review]
@PhilGrayson Ok, after talking to a few of the other maintainers I think we are ok with this as is. You are totally correct that we have always said in the docs that we will not delete CRDs and deleting CRDs is very dangerous. We also want to follow the "principle of least surprise" for all Helm users, so that doing something like migrating CRDs doesn't accidentally result in deleted CRDs for someone.

We are going to run this by the other maintainers in the dev call this week before merging,

**Review by bacongobbler — APPROVED** [review]

**Review by jlegrone — APPROVED** [review]

**Comment by thomastaylor312** [comment]
Everyone in the dev call was good with this, so we are fine with merging. Thanks again for this PR!

**Comment by PhilGrayson** [comment]
That's brilliant news. Thank you for representing the PR in the call.

**Pull request merged**
